Source/swiftlint/Commands/LintCommand.swift

```diff
@@ -55,7 +55,10 @@ struct LintCommand: CommandType {
             violations += currentViolations
             reporter.reportViolations(currentViolations, realtimeCondition: true)
         }.flatMap { files in
-            reporter.reportViolations(violations, realtimeCondition: false)
+            if isWarningThresholdBroken(configuration, violations: violations) {
+                violations.append(createThresholdViolation(configuration.warningThreshold!))
+            }
+            reporter.reportViolations(violations, realtimeCondition: true)
```

Comment:
Hey, when it was false the new error would not show up in the reporting. Im guessing because the report was already generated and so it didn't generate again without True.

Happy to change if you know how this could be done differently